src/SonarScanner.MSBuild.PreProcessor/JreCaching/TarGzUnpacker.cs

```diff
@@ -0,0 +1,130 @@
+            {
+                ExtractEntry(tarIn, destinationFullPath, entry);
+            }
+        }
+    }
+
+    // ref https://github.com/icsharpcode/SharpZipLib/blob/ff2d7c30bdb2474d507f001bc555405e9f02a0bb/src/ICSharpCode.SharpZipLib/Tar/TarArchive.cs#L644
+    private void ExtractEntry(TarInputStream tar, string destinationFullPath, TarEntry entry)
+    {
+        var name = entry.Name;
+        if (Path.IsPathRooted(name))
+        {
+            // NOTE:
+            // for UNC names...  \\machine\share\zoom\beet.txt gives \zoom\beet.txt
+            name = name.Substring(Path.GetPathRoot(name).Length);
+        }
+
+        name = name.Replace('/', Path.DirectorySeparatorChar);
+        var destinationFile = Path.Combine(destinationFullPath, name);
+        var destinationFileDirectory = Path.GetDirectoryName(Path.GetFullPath(destinationFile)) ?? string.Empty;
+        var isRootDir = entry.IsDirectory && entry.Name == string.Empty;
+
+        if (!isRootDir && !destinationFileDirectory.StartsWith(destinationFullPath, StringComparison.InvariantCultureIgnoreCase))
+        {
+            throw new InvalidNameException("Parent traversal in paths is not allowed");
+        }
+
+        if (entry.IsDirectory)
+        {
+            directoryWrapper.CreateDirectory(destinationFile);
+        }
+        else
+        {
+            directoryWrapper.CreateDirectory(destinationFileDirectory);
+            using var outputStream = fileWrapper.Create(destinationFile);
+            tar.CopyEntryContents(outputStream);
+            outputStream.Close();
+            try
+            {
+                SetPermissions(operatingSystemProvider, entry, destinationFile);
+            }
+            catch (Exception ex) // TODO: Test this when SetPermissions is extracted
+            {
+                // TODO: Add some logging and inject ILogger
+            }
+        }
+
+        // TODO: Move this into an IFilePermissionProvider
+        [ExcludeFromCodeCoverage]
+        static void SetPermissions(IOperatingSystemProvider operatingSystemProvider, TarEntry source, string destination)
+        {
+            if (operatingSystemProvider.IsUnix())
+            {
+                if (operatingSystemProvider.OperatingSystem() is PlatformOS.Alpine)
+                {
+                    // https://github.com/Jackett/Jackett/blob/master/src/Jackett.Server/Services/FilePermissionService.cs#L27
+                    var process = new Process
+                    {
+                        StartInfo = new ProcessStartInfo
+                        {
+                            RedirectStandardError = true,
+                            UseShellExecute = false,
+                            CreateNoWindow = true,
+                            WindowStyle = ProcessWindowStyle.Hidden,
+                            FileName = "chmod",
+                            Arguments = $"+arwx \"{destination}\"" // TODO: Specify more fine-grained permissinos
```

Comment:
chmod on Alpine is based on busybox and supports octal numbers https://stackoverflow.com/a/57665989 (here is nice manpage for chmod with numbers https://ss64.com/bash/chmod.html).
So this code should be 
```suggestion
                            Arguments = $"""{ Convert.ToString(source.TarHeader.Mode, 8) } "{destination}" """", 
```